python-sdk/nuscenes/eval/panoptic/panoptic_track_evaluator.py

```diff
@@ -259,11 +259,54 @@ def get_ptq(self) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
 
         ptq_all = ((iou - ids) / tp_eps) * (tp / tp_half_fp_half_fn_eps)  # Calculate PTQ of all classes.
         soft_ptq_all = ((iou - soft_ids) / tp_eps) * (tp / tp_half_fp_half_fn_eps)  # Calculate soft-PTQ of all classes.
-        mean_ptq = ptq_all[self.include].mean()  # Mean PTQ over all classes except ignored classes.
-        mean_soft_ptq = soft_ptq_all[self.include].mean()  # Mean soft-PTQ over all classes except ignored classes.
+
+        # Get classes that at least has 1 ground truth instance, and is included in self.include.
+        ground_truths = tp_eps + fn
+        valid_classes = ground_truths > 0.5
+        for i in range(valid_classes.shape[0]):
+            if i not in self.include:
+                valid_classes[i] = False
+
+        # Mean PTQ over all classes except ignored classes.
+        mean_ptq = ptq_all[valid_classes].mean()
+        # Mean soft-PTQ over all classes except ignored classes.
+        mean_soft_ptq = soft_ptq_all[valid_classes].mean()
 
         return mean_ptq, ptq_all, mean_soft_ptq, soft_ptq_all
 
+    def get_motsa(self) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
```

Comment:
<nit> Should this method name be something like `get_mots_acc_and_precision`, since it also calculates `mean_sMOTSP` in addition to `mean_MOTSA` and `mean_sMOTSA`?

Reply:
I think it's fine, they are all from MOTSA metrics family (the same paper,  just some different variants), it's the same as `gte_lstq` also returns `s_assoc`, they are both from the LSTQ metrics paper.